url49584c1d41_Arrem_GGS_tgz-pJ8-com_gamezgalaxy_test_console_MainJ8/src/com/gamezgalaxy/GGS/API/plugin/CommandHandler.java

```diff
@@ -0,0 +1,214 @@
+	 * check for group permissions, if the <b>player</b> can't
+	 * execute the command, the player will be notified. 
+	 * @param player
+	 *              The player executing the command
+	 * @param command
+	 *               The command the player will use
+	 * @param args
+	 *            Any arguments that will be passed to this command
+	 */
+	public void execute(CommandExecutor player, String command, String[] args) {
+		if (find(command) == null) {
+			if (player != null)
+				player.sendMessage("Command not found!");
+			else
+				_server.Log("Command not found!");
+		}
+		Command c = find(command);
+		if (args.length == 1 && args[0].equals(""))
+			args = new String[0];
+		if(c != null)
+		{
+			if (player != null && !player.getGroup().canExecute(c))
+				player.sendMessage("Sorry, you dont have permission to execute this command!");
+			else {
+				_server.Log(player.getName() + " used /" + c.getName() + arrayToString(args));
+				c.execute(player, args);
+			}
+		}
+	}
+	
+	private String arrayToString(String[] args) {
+		String finals = "";
+		for (String s : args) {
+			finals += " " + s;
+		}
+		return finals;
+	}
+	
+	/**
+	 * Add a command to the command list. This will load permissions
+	 * for this command, if no permissions are found for it, it will
+	 * save the default permissions for it.
+	 * @param cmd
+	 *           The command to add.
+	 */
+	public void addCommand(Command cmd) {
+		if (commands.contains(cmd))
+			return;
+		commands.add(cmd);
+		try {
+			loadPermissions();
+			savePermissions();
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+	}
+	
+	/**
+	 * Remove a command from the list
+	 * @param cmd
+	 *           The command to remove
+	 */
+	public void removeCommand(Command cmd) {
+		if (!commands.contains(cmd))
+			return;
+		commands.remove(cmd);
+	}
+	
+	/**
+	 * Remove a command from the list.
+	 * @param name
+	 *            The command name.
+	 */
+	public void removeCommand(String name) {
+		Command c = find(name);
+		if (c == null)
+			return;
+		removeCommand(c);
+	}
+	
+	private void loadPermissions() throws IOException {
+		if (!new File("properties").exists())
+			new File("properties").mkdir();
+		if (!new File("properties/commands.config").exists())
+			makeDefault();
+		FileInputStream fstream = new FileInputStream("properties/commands.config");
+		DataInputStream in = new DataInputStream(fstream);
+		BufferedReader br = new BufferedReader(new InputStreamReader(in));
+		String strLine;
+		while ((strLine = br.readLine()) != null)   {
+			if (strLine.startsWith("#"))
+				continue;
+			String cmdname = strLine.split("\\:")[0];
+			for (Command c : commands) {
+				if (c.getName().equalsIgnoreCase(cmdname)) {
+					c.setPermissionLevel(Integer.parseInt(strLine.split("\\:")[1]));
+					break;
+				}
+			}
+		}
+		in.close();
+	}
+	
+	private void makeDefault() {
+		PrintWriter out = null;
+		try {
+			new File("properties/commands.config").createNewFile();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

`File.createNewFile` returns `false` if the named file already exists and an empty file is not created. Check the value returned by `createNewFile()`. Handle when the method returns `false`. Otherwise, you might silently overwrite an existing file which could lead to data loss. See the following diffs on GitHub that make this code change: [diff1](https://github.com/lingochamp/FileDownloader/commit/65069c18a4b1b4cfa758173ef49d69c6897fc77a#diff-2b6ad96df9bf9828c22f961e9e9a5b933ab2df7b2fb0f12d62c044309e1e2842L331), [diff2](https://github.com/m-szalik/javamail/commit/62b1cb45e32a998efbe6cfa09b06752869f590ae#diff-440ef5498fc2c802274cf136e7bd7d682bbc10c5976afb5af718984e24deced4L45)

Learn more: [File.createNewFile in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#createNewFile--)